### PR TITLE
adding docker support

### PR DESCRIPTION
**Pull request by delcroip**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: "2.4"
```
Comment (plynchnlm) [inline docker-compose.yml:1]: Where does this version number come from?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+services:
+ lform-app:
+    build: .
+    container_name: lform
+    restart: on-failure
+    ports:
+      - "${HTTP:-8080}:4200"
+    healthcheck:
+      test: ["CMD", "cd /app && yarn healthcheck"]
```
Comment (plynchnlm) [inline docker-compose.yml:10]: What does yarn healthcheck do?

File: docker-compose.yml (hunk 1 above)
Comment (delcroip) [reply docker-compose.yml:1]: This is the version of the docker-compose file format, the latest is 3 but not really adapted to test/demo docker

File: docker-compose.yml (hunk 2 above)
Comment (delcroip) [reply docker-compose.yml:10]: After you start the container with `docker-compose up -d ` when you do `docker-compose ps` it says if the container is starting or up, for some reason it in that case it says unhealthy but without concequences. 

Of couse you can use docker iso docker-compose but the cli is less user friendly that why i always add the compose file to the Dockerfile.

File: docker-compose.yml (hunk 1 above)
Comment (akanduru) [reply docker-compose.yml:1]: I am not familiar with docker-compose. Has directory /app no relation to docker's WORKDIR?

**Comment by plynchnlm** [comment]
My team members are wondering why a docker container is needed, since lforms is a widget not an app.  It looks like the Docker files expose the test application used for the test code, and the test application was never intended to be a product that someone would use.

**Comment by delcroip** [comment]
I understand but it was not clear to me before I tested it, as I created to docker to test it I did a PR in case it could help other.

I will do the same for the smart app

**Comment by plynchnlm** [comment]
I will see if I can make things more clear in the README, and will close this PR.

**Pull request closed without merging**